### ISSUE TITLE
Issues with preprocessor support disabled

### ISSUE THREAD

**Issue by tomazsustar** [robotpy/cxxheaderparser#60]
### Problem description

In our codebase there are some preprocessor directives almost in every header. After the recent changes parsing of the headers is failing. Would it be possible to ingore the preprocessor directives instead of throwing exceptions? This would eliminate the need to run a preprocessor for every simple header wich unfortunately has some preprocessor directives.

**Comment by virtuald** [comment]
In general, ignoring errors is a very bad practice and leads to inconsistent user experience, and the only reason I was doing it was to carry over some of the bad behavior from CppHeaderParser. Most of the bugs we've received have been effectively "this preprocessor related thing isn't working", so it seems better to me to just disable it entirely to lessen user confusion. 

Using pcpp is pretty easy and has little overhead. In my header2whatever project, we use it (https://github.com/virtuald/header2whatever/blob/master/header2whatever/preprocess.py)... though I am noticing that apparently I had implemented something to ensure that only the included file was remaining.

... probably it would make sense to add this as a separate function for cxxheaderparser?

**Comment by tomazsustar** [comment]
We are on the same page, that ignoring errors is a bad practice. And I agree that supporting preprocessor is tricky and pcpp does it already. I am not suggesting to ignore errors, I am suggesting to make cxxheaderparser to be able to cope with a few more preprocessor directives, than just "#include". I think it would make the cxxheaderparser  more usable.

For examle, I have used pcpp myself to work around those issues. Then I run into errors with the #line directive. I had to override the default setting line_directive of the pcpp to "//line" so that the cxxheaderparsed could cope with the output. I think cxxheaderparser should be able to cope with the #line directive. Maybe it could even make use of it for better error messages.

I think also that #warning could be such candidate.

I'm not sure I'd make a hard dependency to pcpp. Maybe this could be covered just by providing some examples. I leave this decision to you.

Accepting #line and #warning directives already solves most of my concerns.
